Add initial files

Hasan24642 opened the pull request:
- Added entity classes for database tables (House, Store, etc.).
- Implemented CRUD repositories for managing database interactions.
- Utilities for JSON file handling and Hibernate session management.
- Added test classes: JsonUtilsTest and GenerateJsonFilesTest.

Hasan24642 committed:
Add initial files

- Added entity classes for database tables (House, Store, etc.).
- Implemented CRUD repositories for managing database interactions.
- Utilities for JSON file handling and Hibernate session management.
- Added test classes: JsonUtilsTest and GenerateJsonFilesTest.

deserted-operative commented:
Looks amazing so far.
Would it be ok to hold off on merging this into `main` until I’ve containerised it and added a GitHub action to run the unit tests?

Hasan24642 commented:
> Looks amazing so far. Would it be ok to hold off on merging this into `main` until I’ve containerised it and added a GitHub action to run the unit tests?

of course 👍

deserted-operative committed:
Added docker-compose.yml

Added a docker compose file to add support for the orchestration  of the SSH Cloud server and the corresponding mysql database server that it accesses.

This involved editing the hibernate configuration file to hard code in the database credentials. _(Hard coding has been chosen as a simplification for the prototyping process.)_

Currently, the docker-compose file only starts the database server - the functionality for it orchestrating the SSH Cloud Docker container is yet to be added.

deserted-operative committed:
Fixed service name in docker-compose.yml

Fixed the name of the service in the file from `postgres` to `mysql`, to accurately reflect the database system being used.

deserted-operative committed:
Debugging database creation and querying

Reduced the number of steps taking place in the Main method to identify where the database creation and querying is failing.

Updated version of Lombok, as it wouldn't build with the previously specified version.

Altered the database schema for debugging purposes.

Also updated the hibernate dialect to a newer version of mysql, as the docker compose file uses the latest available image of mysql for the database server.

deserted-operative commented:
This feature has become a part of the `feature/SSH-Cloud` branch, and as such, am going to close this pull request and recommend deleting the branch, as it's functionality has been encapsulated and furthered in `feature/SSH-Cloud`.
